internal/resource-request-operator/resourceRequest_controller.go

```diff
@@ -41,7 +44,29 @@ func (r *ResourceRequestReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, nil
 	}
 
-	offerErr := r.generateResourceOffer(&resourceRequest)
+	// check if a foreignCluster with an incoming peering
+	// and clusterID == resourceRequest.Spec.ClusterIdentity.ClusterID already exists.
+	foreignClusterList := &discoveryv1alpha1.ForeignClusterList{}
+	err = r.List(ctx, foreignClusterList, client.MatchingLabels{
+		discovery.ClusterIDLabel: resourceRequest.Spec.ClusterIdentity.ClusterID,
+	})
+
+	if err != nil {
+		klog.Errorf("%s -> unable to List foreignCluster with clusterID %s: %s", r.ClusterID, resourceRequest.Spec.ClusterIdentity.ClusterID, err)
+		return ctrl.Result{}, nil
+	}
+
+	if len(foreignClusterList.Items) == 0 {
+		// if does not exist any foreignCluster which satisfies above conditions create a new one.
```

Comment:
IMHO, this comment has limited value, since it mostly repeats what is obvious from the code. In comments you should mostly elaborate on high level logic, why it is needed and why certain conditions can occur (e.g. why the foreigncluster may/may not exist), without repeating low level details.